Applications/xia2_main.py

```diff
@@ -63,6 +65,36 @@ def check_environment():
         )
 
 
+def check_hdf5_master_files(master_files):
+    """Check the input HDF5 master files look a little bit like HDF5 master
+    files and not just the data files: if the latter then sys.exit() with a
+    helpful message"""
+
+    bad = []
+
+    for filename in master_files:
+        with h5py.File(filename, "r") as f:
+            if b"/data" in f and b"/entry" not in f:
+                bad.append(filename)
+
+    if bad:
+
+        dirs = set([os.path.split(b)[0] for b in bad])
+        masters = sum((glob.glob(os.path.join(d, "*_master.h5")) for d in dirs), [])
+        nxss = sum((glob.glob(os.path.join(d, "*.nxs")) for d in dirs), [])
```

Comment:
I wonder if using `pathlib` here would make the code simpler?

Also I would probably prefer `itertools.chain.from_iterable()` to `sum`. Using `sum()` in this way seems like a bit of python dark magic.

Reply:
Furthermore, `sum(list_of_lists, [])` has quadratic complexity, whereas `itertools.chain.from_iterable(list_of_lists)` has linear complexity:
https://mathieularose.com/how-not-to-flatten-a-list-of-lists-in-python/

Reply:
🤷‍♂️ 

Humbly assert that the complexity of `sum` on a short list of short lists is unlikely to have an impact on how long it will take to compose the error message

But I suppose that this is _technically correct_

Reply:
Done as instructed. 

Personally I find this much more cumbersome and much less easier to read